src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -117,8 +120,8 @@ private BoundExpression VisitIndexerAccess(BoundIndexerAccess node, bool isLeftO
                 // This node will be rewritten with MakePropertyAssignment when rewriting the enclosing BoundAssignmentOperator.
 
                 return oldNodeOpt != null ?
-                    oldNodeOpt.Update(rewrittenReceiver, initialBindingReceiverIsSubjectToCloning: ThreeState.Unknown, indexer, arguments, argumentNamesOpt, argumentRefKindsOpt, expanded, argsToParamsOpt, defaultArguments, type) :
-                    new BoundIndexerAccess(syntax, rewrittenReceiver, initialBindingReceiverIsSubjectToCloning: ThreeState.Unknown, indexer, arguments, argumentNamesOpt, argumentRefKindsOpt, expanded, argsToParamsOpt, defaultArguments, type);
+                    oldNodeOpt.Update(rewrittenReceiver, initialBindingReceiverIsSubjectToCloning: ThreeState.Unknown, indexer, arguments, argumentNamesOpt, argumentRefKindsOpt, expanded, AccessorKind.Set, argsToParamsOpt, defaultArguments, type) :
```

Comment:
```oldNodeOpt.AccessorKind```?

Reply:
The value `oldNodeOpt` will be `null` in the case of object initializers. Example:

```csharp
var s =new S()
{
    [0] = 1
};

struct S {
    public int this[int index] 
    {
        get => 0;
        set { }
    }
}
```

The comment and code path pretty much dictates this has to be a `set` (LHS of assignment, not ref returning) hence I felt good about `AccessorKind.Set` here. 

Can add a check that validates if `oldNodeOpt` is not null that it has the `AccessorKind.Set` value.

Reply:
> The value `oldNodeOpt` will be `null` in the case of object initializers.

The comment was specifically for the branch that uses `oldNodeOpt` as a template.